packages/react-menu/src/components/MenuTrigger/useMenuTrigger.ts

```diff
@@ -125,25 +130,21 @@ export const useMenuTrigger_unstable = (props: MenuTriggerProps, ref: React.Ref<
     'aria-haspopup': 'menu',
     'aria-expanded': open,
     id: child?.props?.id || triggerId,
-    onClick,
-    onMouseEnter,
-    onMouseLeave,
-    onKeyDown,
-    onContextMenu,
-    onMouseMove,
+    ref: useMergedRefs(triggerRef, child?.ref),
+    onClick: useMergedEventCallbacks(child?.props?.onClick, onClick),
+    onMouseEnter: useMergedEventCallbacks(child?.props?.onMouseEnter, onMouseEnter),
+    onMouseLeave: useMergedEventCallbacks(child?.props?.onMouseLeave, onMouseLeave),
+    onKeyDown: useMergedEventCallbacks(child?.props?.onKeyDown, onKeyDown),
+    onContextMenu: useMergedEventCallbacks(child?.props?.onContextMenu, onContextMenu),
+    onMouseMove: useMergedEventCallbacks(child?.props?.onMouseMove, onMouseMove),
   };
 
   if (!open && !isSubmenu) {
     overrideProps['aria-expanded'] = undefined;
   }
 
   return {
-    children: useTriggerElement({
-      children,
-      ref: useMergedRefs(triggerRef, ref),
-      outerProps: rest,
-      overrideProps,
-    }),
+    children: applyTriggerPropsToChildren(children, overrideProps),
```

Comment:
As we discussed in offline the behavior of `applyTriggerPropsToChildren()` should be fixed to respect user's input:

```jsx
<Tooltip content="bar" relationship="label">
  <Button aria-label="foo" />
</Tooltip>
```

In this case `button` should have `foo` as `aria-label`, but it has `bar`:

![image](https://user-images.githubusercontent.com/14183168/151793383-b5419c01-188d-4fe9-8333-2ddaf675f36d.png)

---

This change could be done either in this PR or in a follow up.

Reply:
This needs to be handled by the individual trigger components to decide which props to override and which not to. `applyTriggerPropsToChildren` can't do it because for example, we want the merged event handlers on triggerProps to override the child's; but we don't (necessarily?) want the aria props to override them.

I'll handle aria-label for tooltip in a separate PR.

Reply:
We can do something like that:

```tsx
applyTriggerPropsToChildren(children, {
  defaultProps: { /* aria props there */},
  overrideProps: {/* handlers there */}
})
```

In this case, we could handle it in a single place. WDYT?

Reply:
It seems a little overkill IMO, and it's something that the caller could easily do if they wanted:
```ts
applyTriggerPropsToChildren(children, { 
  ...defaultProps, 
  ...child?.props, 
  ...overrideProps,
});
```

Also, it still doesn't help with the case where we don't want to set `aria-label` on the child if it has `aria-labelledby`.

Reply:
```tsx
applyTriggerPropsToChildren(children, { 
  ...defaultProps, // aria props
  ...child?.props, 
  ...overrideProps, // merged event handlers
});
```

> Also, it still doesn't help with the case where we don't want to set aria-label on the child if it has aria-labelledby.

Well, if `overrideProps` will contain only merged event handlers that it will work as expected (user's input wins).